client/components/data/query-whois/index.jsx

```diff
@@ -1,29 +1,27 @@
 import PropTypes from 'prop-types';
-import { Component } from 'react';
-import { connect } from 'react-redux';
+import { useEffect } from 'react';
+import { useDispatch } from 'react-redux';
 import { requestWhois } from 'calypso/state/domains/management/actions';
 import isRequestingWhois from 'calypso/state/selectors/is-requesting-whois';
 
-class QueryWhois extends Component {
-	UNSAFE_componentWillMount() {
-		if ( this.props.isRequesting ) {
-			return;
-		}
-		this.props.requestWhois( this.props.domain );
+const request = ( domain ) => ( dispatch, getState ) => {
+	if ( domain && ! isRequestingWhois( getState(), domain ) ) {
+		dispatch( requestWhois( domain ) );
 	}
+};
 
-	render() {
-		return null;
-	}
+function QueryWhois( { domain } ) {
+	const dispatch = useDispatch();
+
+	useEffect( () => {
+		dispatch( request( domain ) );
+	}, [ dispatch, domain ] );
```

Comment:
The previous component would only dispatch a request on mount, rather than when these props change. Is that ok?

Reply:
That was a bug, it just accidentally never happened in practice. The component should start fetching data for a new domain when the `domain` prop changes.